core/src/components/radio/radio.scss

```diff
@@ -59,3 +61,131 @@ input {
 :host(:focus) {
   outline: none;
 }
+
+:host(.in-item:not(.legacy-radio)) {
+  width: 100%;
+}
+
+// Radio Wrapper
+// --------------------------------------------------
+.radio-wrapper {
+  display: flex;
+
+  position: relative;
+
+  flex-grow: 1;
+
+  align-items: center;
+
+  height: 100%;
+
+  transition: background-color 15ms linear;
+
+  cursor: inherit;
+}
+
+// Radio Label
+// ----------------------------------------------------------------
+
+.label-text-wrapper {
+  /**
+   * This ensures that double tapping this text
+   * clicks the <label> and focuses the radio
+   * when a screen reader is enabled.
+   */
+  pointer-events: none;
+
+  text-overflow: ellipsis;
+
+  white-space: nowrap;
+
+  overflow: hidden;
+}
+
+// Radio Native Wrapper
+// ----------------------------------------------------------------
+
+.native-wrapper {
+  display: flex;
+
+  align-items: center;
+}
+
+// Radio Justify
+// --------------------------------------------------
+
+:host(.radio-justify-space-between) .radio-wrapper {
+  justify-content: space-between;
+}
+
+:host(.radio-justify-start) .radio-wrapper {
+  justify-content: start;
+}
+
+:host(.radio-justify-end) .radio-wrapper {
+  justify-content: end;
+}
+
+// Radio Label Placement - Start
+// ----------------------------------------------------------------
+
+/**
+ * Label is on the left of the radio in LTR and
+ * on the right in RTL.
+ */
+:host(.radio-label-placement-start) .radio-wrapper {
+  flex-direction: row;
+}
+
+:host(.radio-label-placement-start) .label-text-wrapper {
+  /**
+   * The margin between the label and
+   * the radio should be on the end
+   * when the label sits at the start.
+   */
+  @include margin(0, 8px, 0, 0);
```

Comment:
The label is still visible even if it's empty, leading to an extra bit of white space. (This is especially obvious in the new `a11y` test, which has extra space to the left of the second radio.) You'll want to hide the label element if a label isn't provided; toggle and checkbox use a `label-text-wrapper-hidden` class you could copy: https://github.com/ionic-team/ionic-framework/blob/8f573238909292088969e679a17dc1ff3be97539/core/src/components/checkbox/checkbox.scss#L102-L109